pkg/kots/lint_test.go

```diff
@@ -4098,3 +4103,259 @@ spec:
 		})
 	}
 }
+
+func Test_lintKurlInstaller(t *testing.T) {
+	tests := []struct {
+		name      string
+		specFiles SpecFiles
+		expect    []LintExpression
+	}{
+		{
+			name: "deployment file",
+			specFiles: SpecFiles{
+				{
+					Name: "deployment.yaml",
+					Path: "deployment.yaml",
+					Content: `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: example-nginx
+  labels:
+    app: example
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: example
+  template:
+    metadata:
+      labels:
+        app: example
+    spec:
+      containers:
+        - image: nginx`,
+				},
+			},
+		},
+		{
+			name: "multiple declarations",
+			specFiles: SpecFiles{
+				{
+					Name: "deployment.yaml",
+					Path: "deployment.yaml",
+					Content: `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: example-nginx
+  labels:
+    app: example
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: example
+  template:
+    metadata:
+      labels:
+        app: example
+    spec:
+      containers:
+        - image: nginx
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+spec:
+  containers:
+  - name: nginx
+    image: nginx:1.14.2
+    ports:
+    - containerPort: 80`,
+				},
+			},
+		},
+		{
+			name: "valid installer",
+			specFiles: SpecFiles{
+				{
+					Name: "installer.yaml",
+					Path: "installer.yaml",
+					Content: `apiVersion: cluster.kurl.sh/v1beta1
+kind: Installer
+metadata:
+  name: latest
+spec:
+  kubernetes:
+    version: latest
+  containerd:
+    version: latest
+  weave:
+    version: latest
+`,
+				},
+			},
+		},
+		{
+			name: "invalid installer",
+			specFiles: SpecFiles{
+				{
+					Name: "installer.yaml",
+					Path: "installer.yaml",
+					Content: `apiVersion: cluster.kurl.sh/v1beta1
+kind: Installer
+metadata:
+  name: latest
+spec:
+  kubernetes:
+    version: latest
+  weave:
+    version: latest`,
+				},
+			},
+			expect: []LintExpression{
+				{
+					Rule:    "misconfiguration",
```

Comment:
this rule seems pretty generic to me. could we change it to be more specific? something like `kurl-installer-misconfiguration`?